### PR TITLE
feature/add check department logic

### PR DESCRIPTION
**Pull request by borovHokage**

### CODE REVIEW

File: src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs [hunk 1]
```diff
@@ -35,38 +35,45 @@ public class CreateProjectCommand : ICreateProjectCommand
...
-        private readonly IProjectInfoMapper _projectInfoMapper;
         private readonly IHttpContextAccessor _httpContextAccessor;
-        private readonly IRequestClient<IGetDepartmentRequest> _rcGetDepartment;
         private readonly IRequestClient<ICreateWorkspaceRequest> _rcCreateWorkspace;
         private readonly IRequestClient<ICheckUsersExistence> _rcCheckUsersExistence;
         private readonly IRequestClient<ICreateWorkTimeRequest> _rcCreateWorkTime;
+        private readonly IRequestClient<ICheckDepartmentsExistence> _rcCheckDepartmentsExistence;
 
-        private IGetDepartmentResponse GetDepartment(Guid departmentId, List<string> errors)
+        private List<Guid> CheckDepartmentExistence(Guid? departmentId, List<string> errors)
```
Comment (BelehovEgor) [inline src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs:44]: check existence in validator, if request contains incorrect department id i think we have to throw validation exc, but you create project. I don't think this is good.

**Review by BelehovEgor — CHANGES_REQUESTED** [review]

File: src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs (hunk 1 above)
Comment (borovHokage) [reply src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs:44]: fixed

File: src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs [hunk 2]
```diff
@@ -35,38 +35,40 @@ public class CreateProjectCommand : ICreateProjectCommand
...
-                    IGetDepartmentRequest.CreateObj(null, departmentId), timeout: TimeSpan.FromSeconds(2)).Result;
-
+                var response = _rcCheckDepartmentsExistence.GetResponse<IOperationResult<ICheckDepartmentsExistence>>(
+                    ICheckDepartmentsExistence.CreateObj(new List<Guid> { departmentId.Value })).Result;
                 if (response.Message.IsSuccess)
                 {
-                    return response.Message.Body;
+                    if (!response.Message.Body.DepartmentIds.Any())
+                    {
+                        errors.Add("Department Id does not exist");
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs:57]: add id value to error response

File: src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs [hunk 3]
```diff
@@ -209,24 +209,12 @@ public OperationResultResponse<ProjectInfo> Execute(ProjectRequest request)
...
-                department = GetDepartment(request.DepartmentId.Value, response.Errors);
-
-                if (!response.Errors.Any() && department == null)
-                {
-                    throw new BadRequestException("Project department not found.");
-                }
-            }
+            List<Guid> existDepartments = CheckDepartmentExistence(request.DepartmentId, response.Errors);
 
             Guid userId = _httpContextAccessor.HttpContext.GetUserId();
```
Comment (MariyaSin) [inline src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs:214]: remove to mapper

Comment (borovHokage) [reply src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs:210]: discussed

File: src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs (hunk 2 above)
Comment (borovHokage) [reply src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs]: fixed

**Review by MariyaSin — APPROVED** [review]

**Review by Cheephot — APPROVED** [review]

**Review by BelehovEgor — APPROVED** [review]

**Review by VLiamin — APPROVED** [review]

**Pull request merged**